bin/oe/rpc.rs

```diff
@@ -189,44 +244,84 @@ pub fn new_ws<D: rpc_apis::Dependencies>(
         }
         false => None,
     };
-    let start_result = rpc_servers::start_ws(
-        &addr,
-        handler,
-        allowed_origins,
-        allowed_hosts,
-        conf.max_connections,
-        rpc::WsExtractor::new(path.clone()),
-        rpc::WsExtractor::new(path.clone()),
-        rpc::WsStats::new(deps.stats.clone()),
-        conf.max_payload,
-    );
-
-    //	match start_result {
-    //		Ok(server) => Ok(Some(server)),
-    //		Err(rpc::ws::Error::Io(rpc::ws::ErrorKind::Io(ref err), _)) if err.kind() == io::ErrorKind::AddrInUse => Err(
-    //			format!("WebSockets address {} is already in use, make sure that another instance of an Ethereum client is not running or change the address using the --ws-port and --ws-interface options.", url)
-    //		),
-    //		Err(e) => Err(format!("WebSockets error: {:?}", e)),
-    //	}
-    match start_result {
-		Ok(server) => Ok(Some(server)),
-		Err(rpc::ws::Error::WsError(ws::Error {
-			                            kind: ws::ErrorKind::Io(ref err), ..
-		                            })) if err.kind() == io::ErrorKind::AddrInUse => Err(
-			format!("WebSockets address {} is already in use, make sure that another instance of an Ethereum client is not running or change the address using the --ws-port and --ws-interface options.", url)
-		),
-		Err(e) => Err(format!("WebSockets error: {:?}", e)),
-	}
+
+    let mut servers = Vec::default();
+
+    servers.push(format_ws_error(
+        &url,
+        rpc_servers::start_ws(
+            &addr,
+            handler,
+            allowed_origins.clone(),
+            allowed_hosts.clone(),
+            conf.max_connections,
+            rpc::WsExtractor::new(path.clone()),
+            rpc::WsExtractor::new(path.clone()),
+            rpc::WsStats::new(deps.stats.clone()),
+            conf.max_payload,
+        ),
+    )?);
```

Comment:
If `format_ws_error` arguments to be changed, then it is possible just to use `map_err` for the result of `start_ws`. Also, if code duplication could be eliminated, then we may get rid of `format_ws_error` function completely